Should not define TAG in code file

### IDE / Tooling

PlatformIO

### What happened?

Using Arduino framework with Platformio, configured to use:
build_flags =
    -D TAG=\"MYTAG\"
    -D USE_ESP_IDF_LOG

Outputs:
`.pio/libdeps/debug/AsyncTCP/src/AsyncTCP.cpp:26: warning: "TAG" redefined
 #define TAG "AsyncTCP"`

Because
https://github.com/ESP32Async/AsyncTCP/blob/d7577f9afc3ed12561997a54d088a6f35e52e4a7/src/AsyncTCP.cpp#L26C1-L26C23

Should use static char* TAG = "AsyncTCP" and not leak #define's.



### Stack Trace

.pio/libdeps/debug/AsyncTCP/src/AsyncTCP.cpp:26: warning: "TAG" redefined
 #define TAG "AsyncTCP"

### Minimal Reproductible Example (MRE)

Use in platform.ini

build_flags =
    -D TAG=\"MYTAG\"
    -D USE_ESP_IDF_LOG

### I confirm that:

- [x] I have read the documentation.
- [x] I have searched for similar discussions.
- [x] I have searched for similar issues.
- [x] I have looked at the examples.
- [x] I have upgraded to the lasted version of AsyncTCP.

--- a/src/AsyncTCP.cpp
+++ b/src/AsyncTCP.cpp
@@ -23,8 +23,6 @@ extern "C" {
 #include <NetworkInterface.h>
 #endif
 
-#define TAG "AsyncTCP"
-
 // https://github.com/espressif/arduino-esp32/issues/10526
 #ifdef CONFIG_LWIP_TCPIP_CORE_LOCKING
 #define TCP_MUTEX_LOCK()                                \

